trajoptlib/src/error.rs

```diff
@@ -0,0 +1,47 @@
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum TrajoptError {
+    #[error("The solver determined the problem to be overconstrained and gave up.")]
```

Comment:
The fact this duplicates the contents of https://github.com/SleipnirGroup/Sleipnir/blob/main/include/sleipnir/optimization/SolverExitCondition.hpp#L47-L78 isn't ideal.

Reply:
yeah, do you have a better option?

Reply:
See my edited message.